drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/rightpanel/SettingsPresenter.java

```diff
@@ -92,23 +94,38 @@ public void setSaveCommand(Command saveCommand) {
         this.saveCommand = saveCommand;
     }
 
+    @Override
+    public void setSaveEnabled(boolean toSet) {
+        saveEnabled = toSet;
+        if (!saveEnabled) {
+            view.removeSaveButton();
+        } else {
+            view.restoreSaveButton();
+        }
+    }
+
     @Override
     public void onSaveButton(String scenarioType) {
-        simulationDescriptor.setSkipFromBuild(view.getSkipFromBuild().isChecked());
-        switch (ScenarioSimulationModel.Type.valueOf(scenarioType)) {
-            case RULE:
-                saveRuleSettings();
-                break;
-            case DMN:
-                saveDMNSettings();
-                break;
+        if (saveCommand != null) {
+            simulationDescriptor.setSkipFromBuild(view.getSkipFromBuild().isChecked());
+            switch (ScenarioSimulationModel.Type.valueOf(scenarioType)) {
+                case RULE:
+                    saveRuleSettings();
+                    break;
+                case DMN:
+                    saveDMNSettings();
+                    break;
+            }
+            saveCommand.execute();
         }
-        saveCommand.execute();
     }
 
     @Override
     public void reset() {
         view.reset();
+        if (!saveEnabled) {
```

Comment:
A small suggestion: if you use !saveEnabled every time, probably it could worth to use saveDisabled.

Reply:
Hi @dupliaka 
Yes, I know I could do the other way around. But as a matter of "comprehensibility" - i.e. to explicit the meaning of this piece of code - that way it is clear that _!saveEnabled_ is a sort of _specific_ situation that "normally" should not happen (unless in very specific cases).

Reply:
Oh, I didn't realize it this way.